### PR TITLE
replacing set-env with new environment syntax

### PR DESCRIPTION
**Pull request by vsoch**
This is a suggested fix for the update to not use set-env. We probably don't need to push forward with it until users start gettig issues and have a good example to test with it.

Signed-off-by: vsoch <vsochat@stanford.edu>

### CODE REVIEW

**Review by AnandChowdhary — APPROVED** [review]
Looks great!

**Comment by vsoch** [comment]
@AnandChowdhary what do you think about using os.system vs actually deriving that file path in python, opening, and appending to it? I am usually cautious to use os.system.

**Review by david-paulus-saad — APPROVED** [review]

**Comment by flxw** [comment]
So, yes we started having issues with it :D
Is there anything I can do to help this get merged?

File: pull-request.py [hunk 1]
```diff
@@ -41,6 +41,11 @@ def parse_into_list(values):
     return [x.strip() for x in values.split(" ")]
 
 
+def set_env(name, value):
+    """helper function to echo a key/value pair to the environement file"""
+    os.system('echo "%s=%s" >> $GITHUB_ENV' % (name, value))
+
+
```
Comment (flxw) [inline pull-request.py:48]: What about using an implementation like this instead?
```python
def set_env(name, value):
    environment_file_path = os.environ.get('GITHUB_ENV')

    with open(environment_file_path, "a") as environment_file:
        environment_file.write("%s=%s" % (name, value))
```

**Review by flxw — CHANGES_REQUESTED** [review]
Building up on the other comment, I think you can relatively easily replace the os.system call here. Could you kindly check it out?

File: pull-request.py (hunk 1 above)
Comment (vsoch) [reply pull-request.py:48]: Yes this is a much better idea, I'll update the PR with this for you to test!

**Comment by vsoch** [comment]
@david-paulus-saad @AnandChowdhary @flxw I've updated the set_env function to write directly to the file, and rebased with master. Please take another look!

**Review by david-paulus-saad — APPROVED** [review]

**Comment by vsoch** [comment]
@david-paulus-saad do you have a link to an action where you tested so I can take a look?

**Review by flxw — APPROVED** [review]
I just ran this implementation in this repo: https://github.com/flxw/test-pr-action-repo
If you check the action execution history it's nice to see how 1.0.6 is now completely broken due to deprecation but the new implementation works nicely.
What can I help to get this merged?

**Comment by francoisaubert1** [comment]
I would also be happy to see this one merged, I had to set ACTIONS_ALLOW_UNSECURE_COMMANDS to true to keep my pipelines working. Thank you!

**Review by AnandChowdhary — APPROVED** [review]

**Comment by vsoch** [comment]
Thanks everyone! I’ll draft a new release soon.

**Comment by vsoch** [comment]
https://github.com/vsoch/pull-request-action/releases/tag/1.0.12

**Pull request merged**